OutOfSchool/OutOfSchool.WebApi/Services/Elasticsearch/ElasticsearchSynchronizationService.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Nest;
+using OutOfSchool.ElasticsearchData;
+using OutOfSchool.ElasticsearchData.Models;
+using OutOfSchool.Services.Enums;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.WebApi.Extensions;
+using OutOfSchool.WebApi.Models;
+
+namespace OutOfSchool.WebApi.Services
+{
+    /// <summary>
+    /// Implements the operations for synchronization databases.
+    /// </summary>
+    public class ElasticsearchSynchronizationService : IElasticsearchSynchronizationService
+    {
+        private readonly IWorkshopService databaseService;
+        private readonly IEntityRepository<ElasticsearchSyncRecord> repository;
+        private readonly IElasticsearchProvider<WorkshopES, WorkshopFilterES> esProvider;
+        private readonly ILogger<ElasticsearchSynchronizationService> logger;
+
+        public ElasticsearchSynchronizationService(
+            IWorkshopService workshopService,
+            IEntityRepository<ElasticsearchSyncRecord> repository,
+            IElasticsearchProvider<WorkshopES, WorkshopFilterES> esProvider,
+            ILogger<ElasticsearchSynchronizationService> logger)
+        {
+            this.databaseService = workshopService;
+            this.repository = repository;
+            this.esProvider = esProvider;
+            this.logger = logger;
+        }
+
+        public async Task<bool> Synchronize()
+        {
+            {
+                Result result;
```

Comment:
Please move declaration of variable closer to it's 1st usage (line 56). 
For variable declaration wherever possible use `var` - it enforces usage of beforementioned rule (in addition to other benefits like explicitness in type conversion and naming).